### ISSUE TITLE
Drag & Drop in Table View

### ISSUE THREAD

**Issue by svachmic** [ReactiveKit/Bond#366]
Hi,

I was trying to implement a simple table view with simple capability to reorder items in the table view by drag & drop. This feature is available in the very basic UITableViewController implementation out of the box. However, when I bind my tableView to a dataSource (for simplicity, just an ObservableArray<String>) when I set `self.tableView.setEditing(true, animated: false)`, I only get a delete button but not the reorder control with drag and drop feature. When I delete the binding and let it work by the delegate methods of feeding data, it appears and works all well.

My question is - is this intentional? Is drag and drop somehow turned off while using bindings? How should I approach this in order to make it work **with** Bond?

Thank you!

**Comment by svachmic** [comment]
To give a complete example:

1. This is when it **works**:

```
class EditTableViewController: UITableViewController {
    
    let data = MutableObservableArray<String>(["A", "B", "C", "D", "E"])

    override func viewDidLoad() {
        super.viewDidLoad()
        self.tableView.setEditing(true, animated: false)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    // MARK: - Table view data source
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "editCell")!
        
        cell.textLabel?.text = self.data[indexPath.row]
        cell.showsReorderControl = true
        
        return cell
    }
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return data.count
    }

    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        return true
    }

    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
        if editingStyle == .delete {
            // Delete the row from the data source
            tableView.deleteRows(at: [indexPath], with: .fade)
        } else if editingStyle == .insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
        }    
    }

    override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
        self.data.moveItem(from: fromIndexPath.row, to: to.row)
    }

    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
        return true
    }
}
```

2. This when it **doesn't work**:
```
class EditTableViewController: UITableViewController {
    
    let data = MutableObservableArray<String>(["A", "B", "C", "D", "E"])

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.data.bind(to: self.tableView) { conts, indexPath, tableView in
            let cell = tableView.dequeueReusableCell(withIdentifier: "editCell")!
            
            cell.textLabel?.text = conts[indexPath.row]
            cell.showsReorderControl = true
            
            return cell
        }
        
        self.tableView.setEditing(true, animated: false)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
        return true
    }

    override func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCellEditingStyle, forRowAt indexPath: IndexPath) {
        if editingStyle == .delete {
            // Delete the row from the data source
            tableView.deleteRows(at: [indexPath], with: .fade)
        } else if editingStyle == .insert {
            // Create a new instance of the appropriate class, insert it into the array, and add a new row to the table view
        }    
    }

    override func tableView(_ tableView: UITableView, moveRowAt fromIndexPath: IndexPath, to: IndexPath) {
        self.data.moveItem(from: fromIndexPath.row, to: to.row)
    }

    override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
        return true
    }
}
```

**Comment by srdanrasic** [comment]
Hi @svachmic, thanks for the well defined question :)

Binding to a table view will overtake table view's delegate your `EditTableViewController` won't be a delegate any more.

To restore the delegate, you should set `tableView.reactive.delegate.forwardTo = self` (or  `tableView.bnd_delegate.forwardTo = self` in v5).

Can you try and see if it helps?

**Comment by svachmic** [comment]
Thanks for quick answer @srdanrasic 

I just tried `self.tableView.bnd_delegate.forwardTo = self` (I'm on v5) right below defining the binding. It didn't help unfortunately.

**Comment by srdanrasic** [comment]
Can you do it before defining the binding? :)

**Comment by svachmic** [comment]
Yes, I did that too, I'm sorry for not mentioning it. No change.

**Comment by srdanrasic** [comment]
Ok, we'll find a way :) I'm looking for solutions now.

**Comment by srdanrasic** [comment]
Ok, there was a bug :) I've just released v5.4.3 with the fix.

Oh, and I meant data source, not the delegate: `self.tableView.bnd_dataSource.forwardTo = self`. But still, please update the dependency :)

**Comment by svachmic** [comment]
Thanks for the update :) however, when I do `self.tableView.bnd_dataSource.forwardTo = self`, it rewrites the binding, so it shows nothing at all. When I assign the bnd_delegate it's the same as before (no change).

**Comment by srdanrasic** [comment]
Damn, there was something wrong with the tag. I've fixed it as v5.4.4. Please check out.

**Comment by svachmic** [comment]
Haha, that happens to all of us :) it works great now. Thank you for your time and quick action! Loving ReactiveKit + Bond so far!
